Disable toggling pvp with an outsider in your town

Articdive opened the pull request:
Disabled toggling pvp while an outsider (someone not in your town) is in your town,
Added method getTown(UUID) & getNation(UUID)

LlmDl commented on src/com/palmergames/bukkit/towny/command/TownCommand.java:
```diff
@@ -709,10 +735,23 @@ public void townToggle(Player player, String[] split) throws TownyException {
-				town.setPVP(!town.isPVP());
-				TownyMessaging.sendTownMessage(town, String.format(TownySettings.getLangString("msg_changed_pvp"), town.getName(), town.isPVP() ? "Enabled" : "Disabled"));
-
+				boolean outsiderintown = false;
```
Needs an 
if(TownySettings.configsettinghere) {

}
so that the new code only runs when the new config option is true. (It should be made false by default to keep normal pvp toggling in play)

The comments of the new config option should state that using this feature may cause a bit of lag.

LlmDl replied:
Maybe GTOWN_SETTINGS_OUTSIDERS_PREVENT_PVP_TOGGLE

LlmDl commented on src/com/palmergames/bukkit/config/ConfigNodes.java:
```diff
@@ -455,6 +455,12 @@
 	),
+	GTOWN_SETTINGS_OUTSIDERS_PREVENT_PVP_TOGGLE(
+			"global_town_settings.outsiders_prevent_pvp_toggle",
+			"true",
```
Change to false, we want this feature off by default.

Articdive replied:
Oversaw this, yeah ok
 all good

LlmDl commented on src/com/palmergames/bukkit/config/ConfigNodes.java:
```diff
@@ -455,6 +455,12 @@
+			"global_town_settings.outsiders_prevent_pvp_toggle",
+			"true",
+			"# If Towny should allow the toggling of pvp, while others ,who are not part of the town, are in the town",
+			"# If this option is set to false, it may cause a small amount of lag, when /t toggle pvp is run."
```
"# When active this feature can cause a bit of lag when the /t toggle pvp command is used, depending on how many players are online."

LlmDl commented on src/com/palmergames/bukkit/towny/command/TownCommand.java:
```diff
@@ -710,9 +738,27 @@ public void townToggle(Player player, String[] split) throws TownyException {
-				TownyMessaging.sendTownMessage(town, String.format(TownySettings.getLangString("msg_changed_pvp"), town.getName(), town.isPVP() ? "Enabled" : "Disabled"));
-
+				boolean outsiderintown = false;
+				if (!TownySettings.getOutsidersPreventPVPToggle()) {
```
if (TownySettings.getOutsidersPreventPVPToggle()) { 

- remove !

Articdive replied:
Since we use prevent then we want the config option to be false, right if we want to keep the default values and this as an added feature people can enable?

Articdive replied:
By changing the wording to prevent we would need to remove the ! but then also change the config option to false

Articdive replied:
Which we can do there's no problem just that the wording is a little bit messy if we change it

Articdive commented:
Removed the ! set the default to false,
Problem was: I changed the wording without actually reading the context, which caused it to make no sense
Updated the config messages to those that you suggested
